exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -204,6 +204,8 @@ private ExecConstants() {
   public static final String SERVICE_KEYTAB_LOCATION = SERVICE_LOGIN_PREFIX + ".keytab";
   public static final String KERBEROS_NAME_MAPPING = SERVICE_LOGIN_PREFIX + ".auth_to_local";
 
+  /* Provide resiliency on web server for queries submitted via HTTP */
+  public static final String HTTP_QUERY_FAIL_LOW_HEAP_THRESHOLD = "drill.exec.http.query.fail.low_heap.threshold";
```

Comment:
I would have just put this as a constant in the QueryWrapper class. I don't expect the user to ever modify this and when the QueryWrapper is updated to address large result sets, then we don't have to worry about removing this.

Reply:
I am inclined towards having this as an drill-override.conf property, just so that we have a tuning mechanism. But 85% is a reasonable threshold, so I'll hard-code it as a constant within QueryWrapper.